packages/front-end/components/DecisionCriteria/DecisionCriteriaModal.tsx

```diff
@@ -0,0 +1,638 @@
+    };
+
+    // If we have an ID, we're updating an existing decision criteria
+    if (decisionCriteria?.id) {
+      try {
+        await apiCall(`/decision-criteria/${decisionCriteria.id}`, {
+          method: "PUT",
+          body: JSON.stringify(updatedCriteria),
+        });
+      } catch (error) {
+        console.error("Error updating decision criteria:", error);
+      }
+    } else {
+      // Otherwise, we're creating a new one
+      try {
+        await apiCall("/decision-criteria", {
+          method: "POST",
+          body: JSON.stringify(updatedCriteria),
+        });
+      } catch (error) {
+        console.error("Error creating decision criteria:", error);
+      }
+    }
+    mutate();
+  };
+  // Only render the modal if it's open
+  if (!open) return null;
+
+  const selectSize = "2";
+
+  return (
+    <Modal
+      open={open}
+      header="Modify Decision Criteria"
+      subHeader="Define rules for automatic decision making based on experiment results"
+      close={onClose}
+      submit={!disabled ? handleSave : undefined}
+      cta={!disabled ? "Save Decision Criteria" : undefined}
+      size="lg"
+      trackingEventModalType="decision_criteria_create"
+      trackingEventModalSource={trackingEventModalSource}
+    >
+      <Flex direction="column" gap="2">
+        <Flex direction="column" gap="1">
+          <Text weight="bold" size="2">
+            Name
+          </Text>
+          <div className="form-group">
+            <input
+              type="text"
+              className="form-control"
+              placeholder="Decision Criteria Name"
+              value={form.watch("name")}
+              onChange={(e) => form.setValue("name", e.target.value)}
+              required
+              disabled={disabled}
+            />
+          </div>
+          <Text weight="bold" size="2">
+            Description
+          </Text>
+          <div className="form-group">
+            <textarea
+              className="form-control"
+              placeholder="(optional)"
+              value={form.watch("description")}
+              onChange={(e) => form.setValue("description", e.target.value)}
+              rows={2}
+              disabled={disabled}
+            />
+          </div>
+        </Flex>
+
+        <Text weight="bold" size="2">
+          Rules
+        </Text>
+        <Text size="2">
+          Rules are evaluated in order. If a rule matches, an action is
+          recommended and no further rules are evaluated.
+        </Text>
+
+        {form.watch("rules").map((rule, ruleIndex) => (
+          <Flex
+            key={rule.id}
+            className="appbox mb-1 mt-1"
+            direction="column"
+            gap="2"
+            p="2"
+          >
+            <Flex justify="between" align="center" mb="1">
+              <Text weight="bold" size="2">
+                Rule {ruleIndex + 1}
+              </Text>
+              {ruleIndex > 0 && !disabled && (
```

Comment:
nit: Should we check for overall length of rules to allow users to remove rule 1 and keep rule 2 for example?